### PR TITLE
[HIVEMALL-221] Add cache to reduce Maven build time on Travis CI

### PR DESCRIPTION
**Pull request by chezou**
## What changes were proposed in this pull request?

This PR will speed up the build time on Travis CI. Travis CI will timeout when the build time over 50 minutes (See [documentation](https://docs.travis-ci.com/user/customizing-the-build#build-timeouts)).

## What type of PR is it?

Improvement

## What is the Jira issue?

https://issues.apache.org/jira/browse/HIVEMALL-221

### CODE REVIEW

**Comment by myui** [comment]
Seems not working.. 

`timeout: 1000` helps (?)
https://docs.travis-ci.com/user/caching/#setting-the-timeout

Please add `[HIVEMALL-221]` to the PR title.

**Comment by chezou** [comment]
I'm not sure the exact timing of caching, though, I'd add `timeout` on travis.yml.

**Comment by chezou** [comment]
Same issue has been started from this September https://github.com/travis-ci/travis-ci/issues/10108

**Comment by myui** [comment]
@maropu 

Is this `clean` required?
https://github.com/apache/incubator-hivemall/blob/master/bin/run_travis_tests.sh#L42

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 sudo: false
 
+cache:
+  timeout: 1500
+  directories:
+  - $HOME/.m2
```
Comment (myui) [inline .travis.yml]: Isn't `$HOME/.m2/repository` ?

https://github.com/apache/kafka/blob/trunk/.travis.yml#L52
https://github.com/airlift/drift/blob/master/.travis.yml#L11
https://github.com/mesos/storm/blob/master/.travis.yml#L6

File: .travis.yml [hunk 2]
```diff
@@ -35,7 +40,7 @@ notifications:
   email: false
 
 script:
-  - ./bin/run_travis_tests.sh
+  - travis_wait 10 ./bin/run_travis_tests.sh
```
Comment (myui) [inline .travis.yml]: plz revert this change because this does not effect

**Comment by myui** [comment]
See what happens.

**Comment by myui** [comment]
```
[WARNING] Could not transfer metadata org.apache.hivemall:hivemall-spark2.1:0.5.1-incubating-SNAPSHOT/maven-metadata.xml from/to apache.snapshots (https://repository.apache.org/snapshots): Connect to repository.apache.org:443 [repository.apache.org/207.244.88.140] failed: Connection timed out (Connection timed out)
[WARNING] Failure to transfer org.apache.hivemall:hivemall-spark2.1:0.5.1-incubating-SNAPSHOT/maven-metadata.xml from https://repository.apache.org/snapshots/ was cached in the local repository, resolution will not be reattempted until the update interval of apache-snapshots has elapsed or updates are forced. Original error: Could not transfer metadata org.apache.hivemall:hivemall-spark2.1:0.5.1-incubating-SNAPSHOT/maven-metadata.xml from/to apache-snapshots (https://repository.apache.org/snapshots/): Connect to repository.apache.org:443 [repository.apache.org/207.244.88.140] failed: Connection timed out (Connection timed out)
[WARNING] Failure to transfer org.apache.hivemall:hivemall-spark2.1:0.5.1-incubating-SNAPSHOT/maven-metadata.xml from https://repository.apache.org/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of apache.snapshots has elapsed or updates are forced. Original error: Could not transfer metadata org.apache.hivemall:hivemall-spark2.1:0.5.1-incubating-SNAPSHOT/maven-metadata.xml from/to apache.snapshots (https://repository.apache.org/snapshots): Connect to repository.apache.org:443 [repository.apache.org/207.244.88.140] failed: Connection timed out (Connection timed out)
[INFO] Downloading from apache-snapshots: https://repository.apache.org/snapshots/org/apache/hivemall/hivemall-spark2.1/0.5.1-incubating-SNAPSHOT/hivemall-spark2.1-0.5.1-incubating-SNAPSHOT-sources.jar
[INFO] Downloading from apache.snapshots: https://repository.apache.org/snapshots/org/apache/hivemall/hivemall-spark2.1/0.5.1-incubating-SNAPSHOT/hivemall-spark2.1-0.5.1-incubating-SNAPSHOT-sources.jar
```

hmm.... could we provide mirror repository in travis-ci ?

**Comment by chezou** [comment]
Per this discussion https://github.com/travis-ci/travis-ci/issues/10108,  it seems Apache snapshot repo server changed their policy and it eventually blocks travis-ci jobs. I don't know it's a travis issue or Apache server's one.
